src/ledger.rs

```diff
@@ -149,13 +149,20 @@ pub fn extract_keys_and_template(policy: &str) -> Result<(String, Vec<WalletPubK
     let re = Regex::new(r"((\[.+?\])?[xyYzZtuUvV]pub[1-9A-HJ-NP-Za-km-z]{79,108})").unwrap();
     let mut descriptor_template = policy.to_string();
     let mut pubkeys: Vec<WalletPubKey> = Vec::new();
-    for (index, capture) in re.find_iter(policy).enumerate() {
+    for capture in re.find_iter(policy) {
         let pubkey =
             WalletPubKey::from_str(capture.as_str()).map_err(|_| HWIError::UnsupportedInput)?;
-        if !pubkeys.contains(&pubkey) {
+        let pubkey_index = if !pubkeys.contains(&pubkey) {
             pubkeys.push(pubkey);
-        }
-        descriptor_template = descriptor_template.replace(capture.as_str(), &format!("@{}", index));
+            pubkeys.len() - 1
+        } else {
+            pubkeys
+                .iter()
+                .position(|p| p == &pubkey)
+                .expect("Just checked it's in.")
```

Comment:
Inefficient but hey i can't use a hashmap and also the number of pubkeys shouldn't be too large either.